HDDS-738. Removing REST protocol support from OzoneClient

elek opened the pull request:
Since we have functional {{S3Gateway}} for Ozone which works on REST protocol, having REST protocol support in OzoneClient feels redundant and it will take a lot of effort to maintain it up to date.
As S3Gateway is in a functional state now, I propose to remove REST protocol support from OzoneClient.

Once we remove REST support from OzoneClient, the following will be the interface to access Ozone cluster
 * OzoneClient (RPC Protocol)
 * OzoneFS (RPC Protocol)
 * S3Gateway (REST Protocol)

See: https://issues.apache.org/jira/browse/HDDS-738

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 43 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 5 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 44 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for branch |
| +1 | mvninstall | 574 | trunk passed |
| +1 | compile | 374 | trunk passed |
| +1 | checkstyle | 65 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 721 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 151 | trunk passed |
| 0 | spotbugs | 422 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 616 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 33 | Maven dependency ordering for patch |
| +1 | mvninstall | 533 | the patch passed |
| +1 | compile | 368 | the patch passed |
| +1 | javac | 95 | hadoop-hdds in the patch passed. |
| +1 | javac | 273 | hadoop-ozone generated 0 new + 5 unchanged - 3 fixed = 5 total (was 8) |
| -0 | checkstyle | 39 | hadoop-ozone: The patch generated 15 new + 0 unchanged - 0 fixed = 15 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 7 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 627 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 85 | hadoop-ozone generated 1 new + 25 unchanged - 1 fixed = 26 total (was 26) |
| +1 | findbugs | 635 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 293 | hadoop-hdds in the patch passed. |
| -1 | unit | 356 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 45 | The patch does not generate ASF License warnings. |
| | | 5951 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.s3.TestOzoneClientProducer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1329 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle xml shellcheck shelldocs |
| uname | Linux e0f3a4b70b0d 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 57f7370 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/1/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/1/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/1/testReport/ |
| Max. process+thread count | 1306 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone hadoop-ozone/client hadoop-ozone/common hadoop-ozone/datanode hadoop-ozone/dist hadoop-ozone/integration-test hadoop-ozone/ozone-manager hadoop-ozone/ozonefs hadoop-ozone/tools U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/1/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

anuengineer commented:
There are some minor issues, otherwise, it looks quite good.
1. Unit test failure looks connected to this patch.
  ```
   [ERROR] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 2.141 s <<< FAILURE! - in org.apache.hadoop.ozone.s3.TestOzoneClientProducer
[ERROR] testGetClientFailure[0](org.apache.hadoop.ozone.s3.TestOzoneClientProducer)  Time elapsed: 0.58 s  <<< FAILURE!
java.lang.AssertionError: 
 Expected to find 'Couldn't create protocol ' but got unexpected exception: java.io.IOException: Couldn't create RpcClient protocol
	at org.apache.hadoop.ozone.client.OzoneClientFactory.getClientProtocol(OzoneClientFactory.java:207)
   
```

2. The Checkstyle issues -- mostly unused imports.
3. One minor JavaDoc.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 13 | https://github.com/apache/hadoop/pull/1329 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/1329 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/2/console |
| versions | git=2.7.4 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented on hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestOzoneClientProducer.java:
```diff
@@ -17,31 +17,30 @@
 import static org.apache.hadoop.ozone.s3.AWSAuthParser.X_AMZ_CONTENT_SHA256;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
```
whitespace:end of line

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 4 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 45 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 71 | Maven dependency ordering for branch |
| +1 | mvninstall | 593 | trunk passed |
| +1 | compile | 382 | trunk passed |
| +1 | checkstyle | 81 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 851 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 177 | trunk passed |
| 0 | spotbugs | 441 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 645 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 40 | Maven dependency ordering for patch |
| +1 | mvninstall | 539 | the patch passed |
| +1 | compile | 380 | the patch passed |
| +1 | javac | 101 | hadoop-hdds in the patch passed. |
| +1 | javac | 279 | hadoop-ozone generated 0 new + 5 unchanged - 3 fixed = 5 total (was 8) |
| -0 | checkstyle | 43 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 1 | There were no new shellcheck issues. |
| -1 | whitespace | 0 | The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply |
| +1 | xml | 6 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 759 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 99 | hadoop-ozone generated 1 new + 27 unchanged - 1 fixed = 28 total (was 28) |
| +1 | findbugs | 768 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 347 | hadoop-hdds in the patch passed. |
| -1 | unit | 1465 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 48 | The patch does not generate ASF License warnings. |
| | | 7743 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.TestOzoneConfigurationFields |
|   | hadoop.ozone.om.TestOmAcls |
|   | hadoop.ozone.TestStorageContainerManager |
|   | hadoop.ozone.container.server.TestSecureContainerServer |
|   | hadoop.ozone.om.TestOmBlockVersioning |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1329 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle xml shellcheck shelldocs |
| uname | Linux 8ee0952ffb28 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c92de82 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/3/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/3/artifact/out/whitespace-eol.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/3/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/3/testReport/ |
| Max. process+thread count | 4504 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone hadoop-ozone/client hadoop-ozone/common hadoop-ozone/datanode hadoop-ozone/dist hadoop-ozone/integration-test hadoop-ozone/ozone-manager hadoop-ozone/ozonefs hadoop-ozone/s3gateway hadoop-ozone/tools U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/3/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 93 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 4 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 45 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 76 | Maven dependency ordering for branch |
| +1 | mvninstall | 641 | trunk passed |
| +1 | compile | 397 | trunk passed |
| +1 | checkstyle | 84 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 951 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 207 | trunk passed |
| 0 | spotbugs | 487 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 728 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 41 | Maven dependency ordering for patch |
| +1 | mvninstall | 561 | the patch passed |
| +1 | compile | 392 | the patch passed |
| +1 | javac | 106 | hadoop-hdds in the patch passed. |
| +1 | javac | 286 | hadoop-ozone generated 0 new + 5 unchanged - 3 fixed = 5 total (was 8) |
| +1 | checkstyle | 85 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 1 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 7 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 774 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 110 | hadoop-ozone generated 1 new + 26 unchanged - 1 fixed = 27 total (was 27) |
| +1 | findbugs | 733 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 421 | hadoop-hdds in the patch passed. |
| -1 | unit | 2193 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 51 | The patch does not generate ASF License warnings. |
| | | 8934 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.om.TestOmBlockVersioning |
|   | hadoop.ozone.container.server.TestSecureContainerServer |
|   | hadoop.ozone.om.TestOmAcls |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1329 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle xml shellcheck shelldocs |
| uname | Linux e952495ece17 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 894e230 |
| Default Java | 1.8.0_222 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/4/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/4/testReport/ |
| Max. process+thread count | 4777 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone hadoop-ozone/client hadoop-ozone/common hadoop-ozone/datanode hadoop-ozone/dist hadoop-ozone/integration-test hadoop-ozone/ozone-manager hadoop-ozone/ozonefs hadoop-ozone/s3gateway hadoop-ozone/tools U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/4/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 39 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 4 | No case conflicting files found. |
| 0 | shelldocs | 1 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 48 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 80 | Maven dependency ordering for branch |
| +1 | mvninstall | 617 | trunk passed |
| +1 | compile | 380 | trunk passed |
| +1 | checkstyle | 81 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 857 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 176 | trunk passed |
| 0 | spotbugs | 439 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 644 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 40 | Maven dependency ordering for patch |
| -1 | mvninstall | 287 | hadoop-ozone in the patch failed. |
| -1 | compile | 238 | hadoop-ozone in the patch failed. |
| -1 | javac | 238 | hadoop-ozone in the patch failed. |
| +1 | checkstyle | 84 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 1 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 6 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 731 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 94 | hadoop-ozone generated 1 new + 27 unchanged - 1 fixed = 28 total (was 28) |
| -1 | findbugs | 360 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| +1 | unit | 334 | hadoop-hdds in the patch passed. |
| -1 | unit | 57 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 5944 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.TestOmUtils |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1329 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle xml shellcheck shelldocs |
| uname | Linux 05d3a5ab9fa3 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 6d7f01c |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/5/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/5/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/5/artifact/out/patch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/5/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/5/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/5/testReport/ |
| Max. process+thread count | 428 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone hadoop-ozone/client hadoop-ozone/common hadoop-ozone/datanode hadoop-ozone/dist hadoop-ozone/integration-test hadoop-ozone/ozone-manager hadoop-ozone/ozonefs hadoop-ozone/s3gateway hadoop-ozone/tools U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/5/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 70 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 5 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 48 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 81 | Maven dependency ordering for branch |
| +1 | mvninstall | 629 | trunk passed |
| +1 | compile | 380 | trunk passed |
| +1 | checkstyle | 82 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 865 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 177 | trunk passed |
| 0 | spotbugs | 442 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 647 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 40 | Maven dependency ordering for patch |
| +1 | mvninstall | 550 | the patch passed |
| +1 | compile | 379 | the patch passed |
| +1 | javac | 102 | hadoop-hdds in the patch passed. |
| +1 | javac | 277 | hadoop-ozone generated 0 new + 5 unchanged - 3 fixed = 5 total (was 8) |
| +1 | checkstyle | 88 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 1 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 6 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 771 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 93 | hadoop-ozone generated 1 new + 26 unchanged - 1 fixed = 27 total (was 27) |
| +1 | findbugs | 643 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 341 | hadoop-hdds in the patch passed. |
| -1 | unit | 57 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 43 | The patch does not generate ASF License warnings. |
| | | 6315 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.TestOmUtils |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1329 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle xml shellcheck shelldocs |
| uname | Linux f9b0b83cb1bc 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d1aa859 |
| Default Java | 1.8.0_222 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/6/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/6/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/6/testReport/ |
| Max. process+thread count | 436 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone hadoop-ozone/client hadoop-ozone/common hadoop-ozone/datanode hadoop-ozone/dist hadoop-ozone/integration-test hadoop-ozone/ozone-manager hadoop-ozone/ozonefs hadoop-ozone/s3gateway hadoop-ozone/tools U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/6/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

elek commented:
/retest

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 43 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 5 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 48 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 474 | Maven dependency ordering for branch |
| +1 | mvninstall | 963 | trunk passed |
| +1 | compile | 442 | trunk passed |
| +1 | checkstyle | 102 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1021 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 239 | trunk passed |
| 0 | spotbugs | 546 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 821 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 43 | Maven dependency ordering for patch |
| +1 | mvninstall | 555 | the patch passed |
| +1 | compile | 389 | the patch passed |
| +1 | javac | 104 | hadoop-hdds in the patch passed. |
| +1 | javac | 286 | hadoop-ozone generated 0 new + 5 unchanged - 3 fixed = 5 total (was 8) |
| +1 | checkstyle | 98 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 9 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 770 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 98 | hadoop-ozone generated 1 new + 26 unchanged - 1 fixed = 27 total (was 27) |
| +1 | findbugs | 696 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 337 | hadoop-hdds in the patch passed. |
| -1 | unit | 62 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 7585 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.TestOmUtils |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1329 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle xml shellcheck shelldocs |
| uname | Linux d73fa55384e0 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / b69ac57 |
| Default Java | 1.8.0_222 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/7/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/7/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/7/testReport/ |
| Max. process+thread count | 473 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone hadoop-ozone/client hadoop-ozone/common hadoop-ozone/datanode hadoop-ozone/dist hadoop-ozone/integration-test hadoop-ozone/ozone-manager hadoop-ozone/ozonefs hadoop-ozone/s3gateway hadoop-ozone/tools U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/7/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 40 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 5 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 48 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 35 | Maven dependency ordering for branch |
| +1 | mvninstall | 614 | trunk passed |
| +1 | compile | 397 | trunk passed |
| +1 | checkstyle | 92 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 785 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 187 | trunk passed |
| 0 | spotbugs | 427 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 626 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 50 | Maven dependency ordering for patch |
| +1 | mvninstall | 539 | the patch passed |
| +1 | compile | 399 | the patch passed |
| +1 | javac | 113 | hadoop-hdds in the patch passed. |
| +1 | javac | 286 | hadoop-ozone generated 0 new + 5 unchanged - 3 fixed = 5 total (was 8) |
| +1 | checkstyle | 101 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 1 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 7 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 694 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 102 | hadoop-ozone generated 1 new + 26 unchanged - 1 fixed = 27 total (was 27) |
| +1 | findbugs | 622 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 315 | hadoop-hdds in the patch failed. |
| -1 | unit | 1836 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 64 | The patch does not generate ASF License warnings. |
| | | 7918 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.block.TestBlockManager |
|   | hadoop.ozone.client.rpc.TestContainerStateMachineFailures |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1329 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle xml shellcheck shelldocs |
| uname | Linux 0e9b74bf9e01 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 66cfa48 |
| Default Java | 1.8.0_222 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/8/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/8/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/8/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/8/testReport/ |
| Max. process+thread count | 4787 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone hadoop-ozone/client hadoop-ozone/common hadoop-ozone/datanode hadoop-ozone/dist hadoop-ozone/integration-test hadoop-ozone/ozone-manager hadoop-ozone/ozonefs hadoop-ozone/s3gateway hadoop-ozone/tools U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/8/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

elek commented:
@nandakumar131 @anuengineer Finally it's ready to commit. All the tests are fixed.

There were two failures:

 * TestSecureContainerServer: Fixed by HDDS-2040
 * TestOzoneClientRetriesOnException: Passed locally:

```
[INFO] -------------------------------------------------------
[INFO] Running org.apache.hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 54.818 s - in org.apache.hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
```

The biggest change in this patch is the following:

Until now the `ozone sh` command converted the bucket/volume objects to the rest representation to print out as json. As the rest classes are removed I changed it to print out json directly the original java classes (in json format). Therefore same of the field names will be changed by this patch.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 77 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 4 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 48 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 70 | Maven dependency ordering for branch |
| +1 | mvninstall | 633 | trunk passed |
| +1 | compile | 435 | trunk passed |
| +1 | checkstyle | 82 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 885 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 191 | trunk passed |
| 0 | spotbugs | 476 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 690 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 40 | Maven dependency ordering for patch |
| +1 | mvninstall | 589 | the patch passed |
| +1 | compile | 394 | the patch passed |
| +1 | javac | 106 | hadoop-hdds in the patch passed. |
| +1 | javac | 288 | hadoop-ozone generated 0 new + 5 unchanged - 3 fixed = 5 total (was 8) |
| +1 | checkstyle | 85 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 7 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 736 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 94 | hadoop-ozone generated 1 new + 27 unchanged - 1 fixed = 28 total (was 28) |
| +1 | findbugs | 670 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 364 | hadoop-hdds in the patch passed. |
| -1 | unit | 2199 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 49 | The patch does not generate ASF License warnings. |
| | | 8652 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestOzoneRpcClientForAclAuditLog |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/9/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1329 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle xml shellcheck shelldocs |
| uname | Linux fa5a2924f78d 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / b1eee8b |
| Default Java | 1.8.0_222 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/9/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/9/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/9/testReport/ |
| Max. process+thread count | 4859 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone hadoop-ozone/client hadoop-ozone/common hadoop-ozone/datanode hadoop-ozone/dist hadoop-ozone/integration-test hadoop-ozone/ozone-manager hadoop-ozone/ozonefs hadoop-ozone/s3gateway hadoop-ozone/tools U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1329/9/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.
